arch/cortex-m/samv71-hal/src/hal.rs

```diff
@@ -197,7 +197,13 @@ impl SystemHal for Hal {
             peripherals.timer.trigger_all_channels();
 
             Ok(())
-        })
+        });
+
+        if config.disable_interrupts_during_setup {
```

Comment:
What about disabling the interrupts before initializing HAL? This would simplify this a bit

Reply:
Note that this function returns in multiple places, so it's much cleaner to just put the init in scope-bound critical section, and then disable the IRQs, instead of putting a check + enable IRQs at every return.

Reply:
ack